src/eva-gram/src/database/input_tool.py

```diff
@@ -0,0 +1,189 @@
+    cur.execute("CALL public.create_experiments();")
+    # Groups table
+    cur.execute("CALL public.create_groups();")
+    # Variables table
+    cur.execute("CALL public.create_variables();")
+    # Observations table
+    cur.execute("CALL public.create_observations();")
+    # Plots table
+    cur.execute("CALL public.create_plots();")
+
+
+def drop_tables(cur):
+    cur.execute("DROP TABLE IF EXISTS owners CASCADE")
+    cur.execute("DROP TABLE IF EXISTS experiments CASCADE")
+    cur.execute("DROP TABLE IF EXISTS plots CASCADE")
+    cur.execute("DROP TABLE IF EXISTS groups CASCADE")
+    cur.execute("DROP TABLE IF EXISTS observations CASCADE")
+    cur.execute("DROP TABLE IF EXISTS variables CASCADE")
+
+
+def load_dataset_to_db(cur):
+    with open(os.path.join(DATASET_PATH, "dataset.json"), 'rb') as dataset:
+        sample_dataset = json.load(dataset)
+    for owner in sample_dataset['owners']:
+        add_user(cur, owner)
+    for experiment in sample_dataset['experiments']:
+        add_experiment(cur, experiment)
+    for plot in (sample_dataset['plots']):
+        add_plot(cur, plot, sample_dataset['observation_dirs'])
+
+
+def get_observation_name(obs_dirs, filename):
+    # TODO: resolve duplicate filenames from different observations
+    for observation in obs_dirs:
+        obs_dir_path = os.path.join(EXPERIMENT_DATA_PATH, observation)
+        for plot in os.listdir(obs_dir_path):
+            if filename == plot:
+                return observation
+    return None
+
+
+def insert_table_record(cur, data, table):
+    query = "INSERT INTO {} (".format(table)
+    query += ', '.join(data)
+    query += ") VALUES ("
+    query += ', '.join(["%s" for i in range(len(data))])
+    query += ")"
+
+    cur.execute(query, tuple(data.values()))
+
+
+def add_user(cur, user_obj):
+    # check not null constraints
+    # TODO: owner_id made optional for user to provide
+    if 'owner_id' not in user_obj or 'username' not in user_obj:
+        print("Skipping current user: needs both owner_id and username")
+        return 1
+    else:
+        insert_table_record(cur, user_obj, "owners")
+    return 0
+
+
+def delete_user(cur, username):
+    cur.execute("DELETE FROM owners WHERE username=%s", (username,))
+
+
+def add_experiment(cur, experiment_obj):
+    # check not null constraints
+    # TODO: experiment_id made optional for user to provide
+    if 'experiment_id' not in experiment_obj or 'owner_id' not in experiment_obj:
+        print("Skipping current experiment: must provide both experiment_id and owner_id")
+        return 1
+    else:
+        insert_table_record(cur, experiment_obj, "experiments")
+    return 0
+
+
+def add_plot(cur, plot_obj, observation_dirs):
+    # check not null constraints
+    # TODO: plot_id made optional for user to provide
+    if 'plot_id' not in plot_obj or 'plot_file' not in plot_obj or 'experiment_id' not in plot_obj or 'group_id' not in plot_obj or 'observation_id' not in plot_obj:
```

Comment:
This might not be relevant once we move to django, but a cleaner way of doing this is could be like:

```
required = ['plot_id', 'plot_file', 'experiment_id', 'group_id', 'observation_id']
difference = set(required).difference(plot_obj)
if len(difference) > 0:
    print(f'Missing required keys for plot: {difference}')
```
and perhaps it could be generalized in `insert_table_record`.